picard/coverart/providers/caa.py

```diff
@@ -109,7 +111,8 @@ def caa_url_fallback_list(desired_size, thumbnails):
                 url = thumbnails.get(size_alias, None)
         if url is not None:
             urls.append(url)
-    return urls
+            widths.append(item_id)
+    return urls, widths
```

Comment:
Here we should rather return a list of named tuples:

```python
CAAThumbnailListItem = namedtuple('CAAThumbnailListItem', ('url', 'width')

...

thumbnail_list = []

...
    thumbnail_list.append(CAAThumbnailListItem(url=url, width=item_id))
return thumbnail_list
```

BTW, I wonder also about `item_id` here, not really a great name (but that's not your fault;).

And below:

```python
                        thumbnail_list = caa_url_fallback_list(config.setting['caa_image_size'], image['thumbnails'])
                        if not thumbnail_list or is_pdf:
                            url = image['image']
                        else:
                            image_data = {'width': thumbnail_list[0].width, 'height': -1}
...
```